typescript/common-resolvers/src/export/index.ts

```diff
@@ -0,0 +1,53 @@
+import { QueryExportDatasetArgs, ExportFormat } from "@labelflow/graphql-types";
+import { v4 as uuidv4 } from "uuid";
+
+import { exportToCoco } from "./format-coco/index";
+import { exportToYolo } from "./format-yolo/index";
+import { Context } from "../types";
+
+const generateExportFile = async (
+  args: QueryExportDatasetArgs,
+  { repository }: Context
+): Promise<Blob> => {
+  switch (args.format) {
+    case ExportFormat.Yolo: {
+      return await exportToYolo(
+        args.where.datasetId,
+        args?.options?.yolo ?? {},
+        {
+          repository,
+        }
+      );
+    }
+    case ExportFormat.Coco: {
+      return await exportToCoco(
+        args.where.datasetId,
+        args?.options?.coco ?? {},
+        {
+          repository,
+        }
+      );
+    }
+    default: {
+      throw new Error("Unsupported format");
+    }
+  }
+};
+
+const exportDataset = async (
+  _: any,
+  args: QueryExportDatasetArgs,
+  { repository }: Context
+) => {
+  const fileExport = await generateExportFile(args, { repository });
+  const outUrl = (await repository.upload.getUploadTargetHttp(uuidv4()))
+    ?.uploadUrl;
+  await repository.upload.put(outUrl, fileExport);
+  return outUrl;
```

Comment:
Isn't there a more straightforward way to this when running in the Service Worker? Let's discuss this during the DSM.